### PR TITLE
libcurl: Curl with additional url to download certificate

### PR DESCRIPTION
**Pull request by roalter**
* Enable the recipe to download an alternative cacert during creation.
* Make  use of a commited cacert or at least ask the environment for CONAN_CA_BUNDLE, CURL_CA_BUNDLE or REQUESTS_CA_BUNDLE (python)

First attempt.

Specify library name and version:  **libcurl/***

**Question:** Why is the cacert.pem needed at all? Why not use the system certificates? 

the file was not used in any other recipe. 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by roalter** [comment]
Searched for usages of **ca_cert.pem** but didn't find them. Any ideas?

**Comment by RubenRBS** [comment]
Hi @roalter thanks for your contribution and your patience while we processed the rest of the PR backlog. I see your idea behind the changes, and I'll send a PR for your branch to get this thru the finish line :)

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 import os
 import re
+import shutil
```
Comment (RubenRBS) [inline recipes/libcurl/all/conanfile.py:15]:
```suggestion
```

File: recipes/libcurl/all/conanfile.py [hunk 2]
```diff
@@ -226,7 +227,17 @@ def build_requirements(self):
...
+        return 
+
+        #data = { "url":
+        #             [
+        #                 "https://curl.se/ca/cacert-2023-01-10.pem"
+        #             ],
+        #         "sha256": "fb1ecd641d0a02c01bc9036d513cb658bbda62a75e246bedbc01764560a639f0"
+        #        }
+        #
+        #download(self, filename="cacert.pem", **data)
```
Comment (RubenRBS) [inline recipes/libcurl/all/conanfile.py:240]: Need to use the global conf

**Review by RubenRBS — CHANGES_REQUESTED** [review]

**Review by uilianries — CHANGES_REQUESTED** [review]
Thank you for your contribution. Indeed makes sense providing an alternative way to pass the certificate, but the current version needs improvements.

Blocking this PR until having a better solution proposed by @RubenRBS

File: recipes/libcurl/all/conanfile.py (hunk 2 above)
Comment (RubenRBS) [inline recipes/libcurl/all/conanfile.py:240]:
```suggestion
        
        cert_url = self.config.get("user.libcurl.cert:url", check_type=str) or "https://curl.se/ca/cacert-2023-01-10.pem"
        cert_sha256 = self.config.get("user.libcurl.cert:sha256", check_type=str) or "fb1ecd641d0a02c01bc9036d513cb658bbda62a75e246bedbc01764560a639f0" 
        download(self, cert_url, "cacert.pem", verify=True, sha256=cert_sha256)
```

**Review by RubenRBS — CHANGES_REQUESTED** [review]

File: recipes/libcurl/all/conanfile.py (hunk 1 above)
Comment (RubenRBS) [inline recipes/libcurl/all/conanfile.py:15]:
```suggestion
```

**Review by RubenRBS — CHANGES_REQUESTED** [review]

**Comment by RubenRBS** [comment]
Closing in favour of https://github.com/conan-io/conan-center-index/pull/17628 as this has some CLA issues, thanks a lot for the initial idea!

**Pull request closed without merging**